truffle/src/main/java/org/jruby/truffle/runtime/subsystems/ThreadManager.java

```diff
@@ -61,8 +58,7 @@ public RubyThread getRootThread() {
      */
     @TruffleBoundary
     public void enterGlobalLock(RubyThread thread) {
-        globalLock.lock();
-        currentThread = thread;
+        currentThread.set(thread);
```

Comment:
Yes that method will go away in the cleanup phase once this is merged.
It has just been a lot more practical to keep a minimal diff until then.